src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -129,8 +165,10 @@ public static bool CopyIfStreamChanged (Stream stream, string destination)
 				if (!string.IsNullOrEmpty (directory))
 					Directory.CreateDirectory (directory);
 
-				MonoAndroidHelper.SetWriteable (destination);
-				File.Delete (destination);
+				if (File.Exists (destination)) {
+					SetWriteableUnchecked (destination);
+					File.Delete (destination);
+				}
```

Comment:
After reworking this PR, so things seem clean _and_ correct. I'm not seeing any gains anymore.

I will close this for now, but leave the branch here for later.